src/ui/libs/DatalensChartkit/ChartKit/components/ChartKitTooltip/ChartKitTooltip.tsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+
+import {Popup} from '@gravity-ui/uikit';
+
+import {generateHtml} from '../../../modules/html-generator';
+import {ATTR_DATA_TOOLTIP_CONTENT} from '../../../modules/html-generator/constants';
+
+export type ChartKitTooltipRef = {
+    checkForTooltipNodes: (container: HTMLDivElement | null) => void;
+};
+type ChartKitTooltipProps = {};
+
+type ChartKitTooltipState = {
+    ref: {current: HTMLElement};
+    content: string;
+    open: boolean;
+};
+type TooltipAnchors = Record<string, ChartKitTooltipState>;
+
+const getNodeIdFromMouseEvent = (e: MouseEvent) => {
+    const id = (e.target && 'id' in e.target && e.target.id) as string | undefined;
+    return id || '';
+};
+
+const getTooltipContent = (value = '') => {
+    let result = value;
+
+    try {
+        const json = JSON.parse(value);
+        result = generateHtml(json);
+    } catch {}
```

Comment:
I think there is no need to swallow an error here - it will be difficult to figure out why the json does not turn into correct markup when using invalid tags, for example